infra/terraform/global/assets/create_etcd_ebs_snapshot/create_etcd_ebs_snapshot.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/aws"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"os"
+	"log"
+)
+
+var (
+	connection = ec2.New(session.New())
+)
+
+
+func main() {
+
+	instanceKeyFilter   := os.Getenv("INSTANCE_KEY")
+	instanceValueFilter := os.Getenv("INSTANCE_VALUE")
+
+	// Filter instances with role set to master
+	instanceInput := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name: 	aws.String(fmt.Sprintf("tag:%s", instanceKeyFilter )),
+				Values: []*string{ aws.String(instanceValueFilter)},
+			},
+		},
+	}
+
+	// A reservation corresponds to a command to start instances
+	// A reservation is what you do to provision instances, while an instance is what you get
+	reservation, err := connection.DescribeInstances(instanceInput)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+
+	// Iterate over reservations
+	for index := range reservation.Reservations {
+
+		// Iterate reservations to access ec2 instances
+		for _, inst := range reservation.Reservations[index].Instances {
+
+			// Iterate over each instance's block device metadata
+			for _, blk := range inst.BlockDeviceMappings {
+
+				// Exclude root volumes.  Root volumes have "Delete on termination" set to true mostly
+				rootDevice := *blk.Ebs.DeleteOnTermination
```

Comment:
Is there any case of a volume which has this flag set to `true` but it's actually not a root volume?

e.g. is there any chance that we don't take the snapshot of a volume we should snapshot?

Reply:
Also, if you want to avoid an indentation level (which I think Andy is very keen to), you may have something like this at the beginning of the `if` block:

```Go
if *blk.Ebs.DeleteOnTermination {
  continue
}
```

I would probably keep the comments anyway for clarity.

Is there a more explicit way to check if a volume is root? (I had I quick look at the documentation, didn't find any).

Reply:
I was going to suggest moving the test for a root device to a function (eg `if isRootDevice(blk) {`). If there is a more explicit way to check, it would be nice to keep it abstracted from the main function.

Reply:
That was my concern Too.  I tried looking for a better eval but at the time this allowed me to move on. 

`kops` uses auto scaling groups which typically have root device set to `delete on termination`. Which is why I went with this.  

I agree this could better. I'll need your brain on this one.

Reply:
I mean, what would be the downside of snapshotting the root volumes as well?

Also, is this operation asynchronous as usual or blocking? If it's asynchronous what is the error about, it's only erroring when adding the request to the queue, the snapshot could potentially still fail later?

Andy: Yeah, I thought about having a small utility function as well, but I though maybe was overkill, but you're right, it would probably read nice ✨ 

Yeah, happy to lend you my brain if you need to 👍